docs/articles/book/concepts/TheWhy.md

```diff
@@ -0,0 +1,39 @@
+* **Lower Total Cost of Delivery**. We lower the cost of future change in our code, which means we lower the total cost of delivery. We have more options open to us, and future changes take less time because we are guided by the tests we've created.
+* **We are human. (At least, last time we checked)**. Humans make mistakes. It's great if we can prove that the code works how we think it'll work.
+* **Triangulating issues -- and proving they're fixed**. Something has gone wrong. How do you pinpoint where that's happened? With tests in place, you can express the issue in terms of one or more failing tests. When all tests pass once more, you can prove that the bug is gone.
+
+And there are the longer-term considerations:
+
+* **Tested code is testable code**. Is our code straight-forward and modular enough to be tested? One sure way to prove this is by testing it.
+* **Thinking about how code will be consumed**. Writing tests forces you to think about calling the code you're writing, which helps you see the surface area and usage of the code. This perspective shift can lead to a better experience for someone consuming your code -- even if that someone is you.
+
+## Why TDD?
+
+TDD -- Test-Driven Development (or sometimes Test-Driven _Design_) -- is the art of writing a test before you write the corresponding piece of production code. Why do that?
+
+* **Timeliness**. Returning to code after we've written it to write tests proves how the program works now. But what about our intent? How can we prove that the program works how we intend it to work? This is where TDD comes in; by writing tests at the time we write the code, we gain additional confidence from testing from our intention all the way through to production code.
+* **Design Drivers**. As we mentioned earlier, some benefits of tests are that they often keep code simpler and more modular, and that they force the developer to test-drive the usage of their code. TDD takes this to the next level -- from early on, you are engaging with your the shape of your code. As you approach the design of your code with TDD, it often becomes easier to spot dependencies as they emerge and make them explicit. By performing these cycles early and often, you gain many opportunities for insight into your code and its usage, and your design will benefit from that.
+
+## Why NUnit?
+
+NUnit is a testing framework for .NET. Like other frameworks -- xUnit, MSTest -- it provides a few core components that compose such a framework:
+
+* Ways to define tests
+* A runner to run the tests (though you can hook into the Visual Studio test runner, the ReSharper test runner, NCrunch, or any other runner of your choice)
+* An assertion library to use during testing (though you can substitute for [FluentAssertions](https://fluentassertions.com/) or the library of your choice)
```

Comment:
I think it is important to point out that NUnit actually have two assertion libraries, the classic one, and the fluent one.  The "That" syntax and its siblings - the constraint model - is NUnits own fluent library.  
(Under the hood, the classic calls on the fluent, but that is for another section I would assume)

Reply:
I'm torn on this. My thinking:

* I want this guide to demonstrate a happy path
* I don't want to confuse people with too many concepts 
* The docs already exist for the classic assertions
* People should probably know they exist
* I'm under the impression that the core team (myself included) finds the constraint based syntax to be the recommended out of the box usage, since it also helps avoid explaining the need to not mix up expected and actual values.

So I'm leaning towards using the constraint syntax everywhere except maybe in a few cases when introducing asserts themselves.

Reply:
I absolutely agree with you about using the constraint model in all examples.  My point was just that the words "constraint model" is an NUnit technical term.  The words others would use, and which match more with how it looks and is used, is Fluent Assertions.  So when I teach testing, I say that "NUnit has its own fluent assertions" as a built-in and preferred syntax.   To me, you may just skip the classical all over :-)